### ISSUE TITLE
Nethereum.Generator.Console doesn't work as docs say

### ISSUE THREAD

**Issue by vckbarth** [Nethereum/Nethereum#493]
I'm trying to use Nethereum.Generator.Console. I'm following the instructions of StandardContract tutorial. And I stuck. For example, tutorial suggest these semantics: 

`Nethereum.Generator.Console generate from-abi -abi StandardContract.abi -o . -ns Sample.Ethereum`

And output claims:

`Unrecognized command or argument 'generate'`

So I've switched to different semantics (using -help):

`Nethereum.Generator.Console gen-fromabi -abi StandardContract.abi -o . -ns Sample.Ethereum`

And output claims this time:

`Unexpected character encountered while parsing value: <. Path '', line 0, position 0.`

Could someone correct tutorial, or at least explain why current version of this tool has very different behavior compare to previous (documented in tutorial)?

Command 'gen-fromproject' doesn't work also.

Please help :)
Bartek

**Comment by juanfranblanco** [comment]
@vckbarth yes we have not released that version yet, please check the CI branch to get it. It should be released soon, nevertheless.

**Comment by vckbarth** [comment]
Ok, I've taken 'Nethereum.Generator.Console' from version 3 and it seems to work, but....

I've generated C# classes from StandardContract.abi. Unfortunately there are several issues, for example:

The type 'Sample.Ethereum.StandardContract.CQS.NameFunction' cannot be used as type parameter 'TEthereumContractFunctionMessage' in the generic type or method 'ContractHandler.QueryAsync<TEthereumContractFunctionMessage, TReturn>(TEthereumContractFunctionMessage, BlockParameter)'. There is no implicit reference conversion from 'Sample.Ethereum.StandardContract.CQS.NameFunction' to 'Nethereum.Contracts.FunctionMessage'. [MyStandardContractProject]

Or:

The type 'Sample.Ethereum.StandardContract.CQS.BalancesFunction' cannot be used as type parameter 'TEthereumContractFunctionMessage' in the generic type or method 'ContractHandler.QueryAsync<TEthereumContractFunctionMessage, TReturn>(TEthereumContractFunctionMessage, BlockParameter)'. There is no implicit reference conversion from 'Sample.Ethereum.StandardContract.CQS.BalancesFunction' to 'Nethereum.Contracts.FunctionMessage'. [MyStandardContractProject]

So my questions:
1. What the hell is TEthereumContractFunctionMessage class? Where can I find definition?
2. Why generator is preparing sorce code that doesn't fit with actual API version? I've fixed several class names and references already.
3. Is there any simple way to finish 'StandardContract' properly?

**Comment by Dave-Whiffin** [comment]
Hi - sorry to hear you're having these issues.  (Although please bear in mind the generator console is pre release - the docs and code aren't consistent because of this).
Please run the command below and send me the output:

```
dotnet tool list -g
```

Dave

**Comment by vckbarth** [comment]
Here is output:

`Package Id                       Version      Commands                   
-------------------------------------------------------------------------
nethereum.generator.console      1.0.1        Nethereum.Generator.Console`

Bartek

**Comment by Dave-Whiffin** [comment]
That version is old and probably references the wrong version of Nethereum.Web3. nuget doesn't yet contain the more recent versions of this tool - so you need to reference myget.

Firstly uninstall the generator tool .
``
dotnet tool uninstall -g Nethereum.Generator.Console
```
Then reinstall using the command below  (this if from CI on myget which Juan mentioned):
```
dotnet tool install -g --add-source https://www.myget.org/F/nethereum/api/v3/index.json --version 1.0.2-CI-20190129-130426 Nethereum.Generator.Console
```

This should let you generate classes cleanly (obviously you're probably best trying this in a clean project first).  

Let me know how you get on?

**Comment by Dave-Whiffin** [comment]
Here's the command line options I used to create a test project, generate the code and ensure it builds.  I am using a windows laptop.
  
```
mkdir Issue493
cd Issue493
copy ..\StandardContract.abi .
dotnet new classlib
dotnet add package Nethereum.Web3
Nethereum.Generator.Console generate from-abi -abi StandardContract.abi -o . -ns MyNamespace
dotnet build
```

**Comment by Dave-Whiffin** [comment]
I am giving it a go on a mac now - I haven't got a convenient Ubuntu to test I am afraid - give me 20 mins and I'll get back to you

**Comment by Dave-Whiffin** [comment]
Apologies - I have unconnected problems with my mac.  I am afraid this will need to wait until tomorrow.  I did some reading that suggested that dotnet tool may have some issues with ubuntu - so if there's a chance you can try a windows pc or a laptop to get you going then it's worth doing.

**Comment by vckbarth** [comment]
I understand. Either way I will try to work this out on Linux - maybe this exercise would be useful whatsoever.

**Comment by Dave-Whiffin** [comment]
I tested on the mac this morning - I followed the same process and code generation worked as expected.  I tested using .Net Core 2.2.102 (it's possible there are some tool fixes in that).  I'll try and make some time next week to set up an ubuntu instance.

**Comment by Dave-Whiffin** [comment]
Tested on ubuntu 18.04 and unfortunately I encountered the same issue as you.  I believe it is a dotnet tool or pack issue related to case sensitivity of the version number. https://github.com/dotnet/cli/issues/9870

I am investigating if we can switch our CI build routine to output a lower case version number  ("CI" becomes "ci").   We are also looking to release this package to nuget shortly (could be days - possible weeks) and when we do that we'll have a proper version number which won't be subject to this issue.  It will also mean that the docs and package will be consistent!
